cmake/templates/OpenCVConfig-CUDA.cmake.in

```diff
@@ -14,6 +17,16 @@ else()
   endif()
 endif()
 
+if(OpenCV_USE_CUDNN)
+    if(NOT CUDNN_FOUND)
+        find_host_package(CUDNN ${OpenCV_CUDNN_VERSION} EXACT REQUIRED)
```

Comment:
CUDNN is internal dependency (like many others).

Don't require CUDNN development files to build user applications.
See [this build](http://pullrequest.opencv.org/buildbot/builders/precommit_custom_linux/builds/2175).

Reply:
The cuDNN handles get exposed to the user from the DNN module includes which allows the user to write a custom layer with a CUDA implementation using cuDNN API.

I think a better way would be to abstract away cuDNN completely so that they only play with `csl::cudnn::Handle` object and `csl::Tensor` without requiring `cudnn.h`.

I'll remove this.